ellama.el

```diff
@@ -2147,6 +2167,41 @@ otherwise prompt user for URL to summarize."
       :false-object nil)
      :same)))
 
+(defun ellama-generate-completion-list ()
+  "Generate completion list from context at point."
+  (mapcar (lambda (s)
+	    (car (reverse (string-split s))))
+	  (plist-get
+	   (json-parse-string
+	    (llm-chat
+	     (or ellama-completion-provider
+		 ellama-provider
+		 (ellama-get-first-ollama-chat-model))
+	     (llm-make-chat-prompt
+	      (if (region-active-p)
+		  (buffer-substring-no-properties (region-beginning) (region-end))
+		(buffer-substring-no-properties (point-min) (point)))
+	      :context ellama-completion-list-prompt-template
+	      :response-format '(:type object :properties
+				       (:completions (:type array :items (:type string)))
+				       :required (completions))))
+
+	    :object-type 'plist
+	    :false-object nil
+	    :array-type 'list)
+	   :completions)))
+
+(defun ellama-complete-at-point (&rest _)
+  "Complete at point using ellama."
+  (interactive)
+  (while-no-input
+    (let ((bounds (bounds-of-thing-at-point 'word)))
+      (when bounds
+	(list (car bounds)
+	      (cdr bounds)
+	      (ellama-generate-completion-list)
```

Comment:
`ellama-generate-completion-list` seems like a "heavy" function.

Since it does some non-trivial work, it would be better to return a reference to a function that conforms to a "completion table" interface, instead of calling it right away.